raft/clock.go

```diff
@@ -55,39 +50,14 @@ func (c *Clock) AfterHeartbeatInterval() <-chan chan struct{} {
 // AfterReconnectTimeout returns a channel that fires after the reconnection timeout.
 func (c *Clock) AfterReconnectTimeout() <-chan chan struct{} { return newClockChan(c.ReconnectTimeout) }
 
-// AfterWaitInterval returns a channel that fires after the wait interval.
-func (c *Clock) AfterWaitInterval() <-chan chan struct{} { return newClockChan(c.WaitInterval) }
-
-// HeartbeatTicker returns a Ticker that ticks every heartbeat.
-func (c *Clock) HeartbeatTicker() *Ticker {
-	t := time.NewTicker(c.HeartbeatInterval)
-	return &Ticker{C: t.C, ticker: t}
-}
-
 // Now returns the current wall clock time.
 func (c *Clock) Now() time.Time { return time.Now() }
 
-// Ticker holds a channel that receives "ticks" at regular intervals.
-type Ticker struct {
-	C      <-chan time.Time
-	ticker *time.Ticker // realtime impl, if set
-}
-
-// Stop turns off the ticker.
-func (t *Ticker) Stop() {
-	if t.ticker != nil {
-		t.ticker.Stop()
-	}
-}
-
 // newClockChan returns a channel that sends a channel after a given duration.
 // The channel being sent, over the channel that is returned, can be used to
 // notify the sender when an action is done.
 func newClockChan(d time.Duration) <-chan chan struct{} {
-	ch := make(chan chan struct{})
-	go func() {
-		time.Sleep(d)
-		ch <- make(chan struct{})
-	}()
+	ch := make(chan chan struct{}, 1)
```

Comment:
There are places where the channel is returned to but it never receives the value so this goroutine stays open indefinitely. It caused a shit ton of goroutines to be left open.